cpp/ql/src/semmle/code/cpp/exprs/BuiltInOperations.qll

```diff
@@ -4,15 +4,15 @@ import semmle.code.cpp.exprs.Expr
  * A C/C++ builtin operation. This is the root QL class encompassing
  * built-in functionality.
  */
-abstract class BuiltInOperation extends Expr {
+class BuiltInOperation extends Expr, @buildin_op {
   override string getCanonicalQLClass() { result = "BuiltInOperation" }
 }
 
 /**
  * A C/C++ built-in operation that is used to support functions with variable numbers of arguments.
  * This includes `va_start`, `va_end`, `va_copy`, and `va_arg`.
  */
-class VarArgsExpr extends BuiltInOperation {
+class VarArgsExpr extends BuiltInOperation, @var_args_expr {
   VarArgsExpr() {
     this instanceof BuiltInVarArgsStart
     or
```

Comment:
If things are working properly, you can now remove this charpred. (Since it doesn't affect the dbscheme, you don't have to recreate the internal PR if you add a commit to this PR with such cosmetic changes.)